flycheck.el

```diff
@@ -5841,7 +5841,12 @@ about TSLint."
                  (flycheck-error-new-at
                   (+ 1 .startPosition.line)
                   (+ 1 .startPosition.character)
-                  'warning .failure
+                  (pcase .ruleSeverity
+                    (`"ERROR"   'error)
+                    (`"WARNING" 'warning)
+                    ;; Default to error
```

Comment:
No need for this comment; I think the code is clear without it :) Also, no need to quote strings in `pcase` patterns.